test/unit/gateways/orbital_test.rb

```diff
@@ -882,6 +917,49 @@ def test_currency_code_and_exponent_are_set_for_profile_authorizations
     assert_success response
   end
 
+  # Successful Authorize (A) through Electronic Check
+  def test_successful_authorize_with_echeck
+    @gateway.expects(:ssl_post).returns(successful_authorize_with_echeck_response)
+
+    assert response = @gateway.authorize(50, @echeck, order_id: '2')
+    assert_instance_of Response, response
+    assert_equal 'Approved', response.message
+    assert_success response
+    assert_equal '5F8E8D2B077217F3EF1ACD3B61610E4CD12954A3;2', response.authorization
+  end
+
+  # Authorization failure due to invalid amount in Electronic Check
+  def test_failed_authorize_with_echeck
+    @gateway.expects(:ssl_post).returns(failed_echeck_for_invalid_amount_response)
+
+    assert response = @gateway.authorize(-1, @echeck, order_id: '9baedc697f2cf06457de78')
+    assert_instance_of Response, response
+    assert_failure response
+    assert_equal 'Error validating amount. Must be numeric, equal to zero or greater [-1]', response.message
+    assert_equal '885', response.params['proc_status']
+  end
+
+  # Successful refund through Electronic Check
+  def test_successful_refund_with_echeck
+    @gateway.expects(:ssl_post).returns(successful_refund_with_echeck_response)
+
+    assert response = @gateway.refund(50, '1;2', @options)
```

Comment:
Is '1;2' the correct param to pass? I don't see the difference between that and the failure below.

Reply:
@jessiagee , please note that this is a unit test, where we produce static response to test the correct rendering of the response. Here the response is already given in the **successful_refund_with_echeck_response**. So, the values are just passed so that the method can be called correctly and produce our static response.

Previous one was an actual remote test case where we hit the server and get the response, so in that case we intentionally gave incorrect value to test the failure (negative) case.

Reply:
Thanks for the clarification!